package/MDAnalysis/analysis/pca.py

```diff
@@ -253,18 +253,29 @@ def transform(self, atomgroup, n_components=None, start=None, stop=None,
         """
         if not self._calculated:
             self.run()
-        # TODO has to accept universe or trajectory slice here
-        if self._atoms.atoms != atomgroup.atoms:
-            warnings.warn('This is a transform for different atom types.')
+
         if isinstance(atomgroup, Universe):
             atomgroup = atomgroup.atoms
 
+        if(self._n_atoms != atomgroup.n_atoms):
+            raise ValueError('PCA transform needs to be fit on same number of'
```

Comment:
`'PCA has been fit for {} atoms. Your atomgroup has {} atoms'.format(self._n_atoms, atomgroup.n_atoms)`